lib/proxy/log.rb

```diff
@@ -13,7 +20,18 @@ def logger
     def self.logger
       log_file = ::Proxy::SETTINGS.log_file
       if log_file.upcase == 'STDOUT'
+        if SETTINGS.daemon
+          puts "Settings log_file=STDOUT and daemon=true are incompatible, exiting..."
+          exit 1
+        end
         logger = ::Logger.new(STDOUT)
+      elsif log_file.upcase == 'SYSLOG'
+        begin
+          logger = ::Syslog::Logger.new 'foreman-proxy'
+        rescue
+          puts "Falling back to STDOUT logger"
+          logger = ::Logger.new(STDOUT)
+        end
       else
         # We keep the last 6 10MB log files
         logger = ::Logger.new(log_file, 6, 1024*1024*10)
```

Comment:
I find initialization logic is a bit convoluted now. I think logger deserves it's own initialization method now that will be executed once, on startup (will have to rely on some dynamic programming or lambdas), which would allow to cleanly deal with config errors or fallbacks.